crates/bls12381/src/curves/g1.rs

```diff
@@ -283,6 +311,34 @@ mod tests {
         expect_eq(cs.num_constraints(), &expect!["5051"]);
     }
 
+    #[test]
+    fn test_collinear() {
+        let mut rng = rand::thread_rng();
+        let a = G1Projective::random(&mut rng);
+        let b = G1Projective::random(&mut rng);
+        let c = a + b;
+        let a = G1Affine::from(a);
+        let b = G1Affine::from(b);
+        let c = G1Affine::from(c);
+
+        let mut cs = TestConstraintSystem::<Fp>::new();
+        let a_alloc = G1Point::alloc_element(&mut cs.namespace(|| "alloc a"), &a).unwrap();
+        let b_alloc = G1Point::alloc_element(&mut cs.namespace(|| "alloc b"), &b).unwrap();
+        let c_alloc = G1Point::alloc_element(&mut cs.namespace(|| "alloc c"), &c).unwrap();
+        let res_alloc = a_alloc
+            .collinearcheck(&mut cs.namespace(|| "a+b-c = 0"), &b_alloc, &c_alloc)
+            .unwrap();
```

Comment:
`res_alloc` is unused so you can remove it:
```suggestion
        a_alloc
            .collinearcheck(&mut cs.namespace(|| "a+b-c = 0"), &b_alloc, &c_alloc)
            .unwrap();
```
(or, if you change the name of the function:
```suggestion
        G1Point::
            .assert_collinear(&mut cs.namespace(|| "a+b-c = 0"), &a_alloc, &b_alloc, &c_alloc)
            .unwrap();
```
)